src/Libraries/RevitNodesUI/SelectionCombo.cs

```diff
@@ -64,6 +64,11 @@ public string SelectedString
             set
             {
                 DropDownNodeModel.SelectedString = value;
+
+                if (DropDownNodeModel.SelectedIndex >= 0)
```

Comment:
is the idea here that in different localizations that the index will be more reliable than the string? Doesn't that undo the purpose of using the string to help when the indicies change between Revit versions? FYI @QilongTang 

Maybe I am misunderstanding.

Reply:
Yes, this would cause regression for cases that category index change across Revit versions. Also if we want to do this in the setter, I would rather we obsolete `SelectedString`, no point keeping the property if not going to use it.

Reply:
Just curious
when changing selected-string can selected-index be changed accordingly,
when changing selected-index can selected-string be changed accordingly? 
is this realized by dynamo core side?

Reply:
@mjkkirschner I think it is the opposite.  SelectedIndex is not reliable between localization or Revit files.  Strings are the same but order of collection is different.  For the drop down nodes in Revit for deserialization, the `SelectedIndex` is first `Set` if it exists in the collection.  Index alone does not guarantee alignment.  The `SelectedString` is `Set`...  It checks the alignment of strings (which are all English strings) and it will Re-`Set` the selected index.  That step is what I am adding here in this PR.

Reply:
(maybe it makes sense to add a property changed handler to the DropDownNodeModel and watch for selected string and index prop updates to centralize this.

Reply:
@saintentropy Thank you for the explanation